main.c

```diff
@@ -25,30 +28,21 @@ int main(void)
     __HAL_RCC_PWR_CLK_ENABLE();
     HAL_PWREx_ControlVoltageScaling(PWR_REGULATOR_VOLTAGE_SCALE2);
 
-    HAL_Delay(100);
+    vTaskDelay(1);
```

Comment:
Can you add a comment on why this is needed?

Reply:
not for main. this will need refactoring later